library/extractor/src/main/java/com/google/android/exoplayer2/metadata/id3/TextInformationFrame.java

```diff
@@ -19,52 +19,76 @@
 
 import android.os.Parcel;
 import android.os.Parcelable;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.android.exoplayer2.MediaMetadata;
 import com.google.android.exoplayer2.util.Util;
+import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /** Text information ID3 frame. */
 public final class TextInformationFrame extends Id3Frame {
-
   @Nullable public final String description;
+
+  /** @deprecated Use {@code values[0]} instead. */
+  @Deprecated
   public final String value;
 
-  public TextInformationFrame(String id, @Nullable String description, String value) {
+  /** The text values of this frame. Will always have at least one element. */
+  @NonNull
+  public final List<String> values;
+
+  public TextInformationFrame(String id, @Nullable String description, @NonNull List<String> values) {
     super(id);
+    if (values.isEmpty()) {
+      throw new IllegalArgumentException("Text information frames must have at least one value");
+    }
+
     this.description = description;
-    this.value = value;
+    this.values = ImmutableList.copyOf(values);
+    this.value = this.values.get(0);
+  }
+
+  /** @deprecated Use {@code TextInformationFrame(String id, String description, String[] values} instead */
+  @Deprecated
+  public TextInformationFrame(String id, @Nullable String description, String value) {
+    this(id, description, Collections.singletonList(value) );
   }
 
   /* package */ TextInformationFrame(Parcel in) {
     super(castNonNull(in.readString()));
     description = in.readString();
-    value = castNonNull(in.readString());
+    values = ImmutableList.copyOf(in.createStringArray());
+    value = values.get(0);
   }
 
   @Override
   public void populateMediaMetadata(MediaMetadata.Builder builder) {
+    // Depending on the context this frame is in, we either take the first value of a multi-value
+    // frame because multiple values make no sense, or we join the values together with a comma
+    // when multiple values do make sense.
```

Comment:
During internal review of this PR, @tonihei asked whether this split of "multiple values makes sense for field X but not field Y" is defined anywhere in the spec - or whether it's just your intuition?

I see that https://id3.org/id3v2.4.0-frames says:
> All text information frames supports multiple strings

Given that, we're wondering if it might be safer to just comma-separate every value when writing into a string field.

For `trackNumber` and other 'transform to a single int' fields we would probably still need to only use the first value, since there's no real way to join multiple values together.

What do you think?

(no need to make any additional changes to this PR, just reply to this comment thread and I'll make the changes to the internal version and they will get propagated back here after it's submitted internally)

Reply:
It's largely intuition from my personal experience @icbaker. Feel free to make it more generalized if you believe it aligns with the specification better.